pkg/webhook/pod/pods.go

```diff
@@ -192,8 +181,67 @@ func (pc *PodAdmissionControl) admitDeletePods(name, namespace string) *v1beta1.
 	if l.IsPD() {
 		pdClient := pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster)
 		return pc.admitDeletePdPods(pod, ownerStatefulSet, tc, pdClient)
+	} else if l.IsTiKV() {
+		pdClient := pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster)
+		return pc.admitDeleteTiKVPods(pod, ownerStatefulSet, tc, pdClient)
 	}
 
 	klog.Infof("[%s/%s] is admit to be deleted", namespace, name)
 	return util.ARSuccess()
 }
+
+// Webhook server receive request to create pod
+// if this pod wasn't member of tidbcluster, just let the request pass.
+// Currently we only check with tikv pod
+func (pc *PodAdmissionControl) AdmitCreatePods(ar admission.AdmissionReview) *admission.AdmissionResponse {
+	pod := &core.Pod{}
+	if err := json.Unmarshal(ar.Request.Object.Raw, pod); err != nil {
+		klog.Errorf("Could not unmarshal raw object: %v", err)
+		return util.ARFail(err)
+	}
+
+	name := pod.Name
+	namespace := pod.Namespace
+
+	klog.Infof("receive admission to %s pod[%s/%s]", "create", namespace, name)
+
+	l := label.Label(pod.Labels)
+	if !(l.IsPD() || l.IsTiKV() || l.IsTiDB()) {
+		klog.Infof("pod[%s/%s] is not TiDB component,admit to create", namespace, name)
+		return util.ARSuccess()
+	}
+
+	tcName, exist := pod.Labels[label.InstanceLabelKey]
+	if !exist {
+		klog.Errorf("pod[%s/%s] has no label: %s", namespace, name, label.InstanceLabelKey)
+		return util.ARFail(fmt.Errorf("pod[%s/%s] has no label: %s", namespace, name, label.InstanceLabelKey))
+	}
+
+	tc, err := pc.tcLister.TidbClusters(namespace).Get(tcName)
+	if err != nil {
+		klog.Errorf("failed get tc[%s/%s],refuse to create pod[%s/%s]", namespace, tcName, namespace, name)
+		return util.ARFail(err)
+	}
+
+	ownerStatefulSet, err := getOwnerStatefulSetForTiDBComponent(pod, pc.stsLister)
+	if err != nil {
+		klog.Errorf("failed to get owner statefulset,refuse to create pod[%s/%s]", namespace, name)
+		return util.ARFail(err)
+	}
+
+	ordinal, err := operatorUtils.GetOrdinalFromPodName(name)
+	if err != nil {
+		return util.ARFail(err)
+	}
+
+	if ownerStatefulSet.Status.CurrentReplicas < 1 && ownerStatefulSet.Status.UpdatedReplicas < 1 && ordinal == 0 {
```

Comment:
What's this case for?

Reply:
For initialization. When we receive request for creating tikv-0, there are 2 cases: 
  1. initialize tikv
  2.upgrading tikv

In second case, we shall end evict leader for tikv-1 and in first case, we shall let the request pass.